src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSourceRequest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using NuGet.Common;
+
+namespace NuGet.Protocol
+{
+    /// <summary>
+    /// A non-cached HTTP request handled by <see cref="HttpSource"/>.
+    /// </summary>
+    public class HttpSourceRequest
+    {
+        public static readonly TimeSpan DefaultRequestTimeout = TimeSpan.FromSeconds(100);
+
+        public HttpSourceRequest(string uri, ILogger log)
+            : this(
+                uri,
+                () => HttpRequestMessageFactory.Create(HttpMethod.Get, uri, log))
```

Comment:
I have modified this file -- let me know if you think more changes are necessary.